include/SoapySDR/Types.hpp

```diff
@@ -173,16 +174,19 @@ namespace Detail {
 template <typename Type>
 typename std::enable_if<std::is_same<Type, bool>::value, Type>::type StringToSetting(const std::string &s)
 {
-    if (s == SOAPY_SDR_TRUE) return true;
-    if (s == SOAPY_SDR_FALSE) return false;
-
-    //zeros and empty strings are false
-    if (s == "0") return false;
-    if (s == "0.0") return false;
-    if (s == "") return false;
-
-    //other values are true
+  if (s.empty() || s == SOAPY_SDR_FALSE) {
+    return false;
+  }
+  if (s == SOAPY_SDR_TRUE) {
     return true;
+  }
+  try {
+    // C++: Float conversion to bool is unambiguous
+    return std::stod(s);
+  } catch (std::invalid_argument &e) {
+  }
```

Comment:
Since we are ignoring the exception contents, leave the `e` out to prevent unused warning (depends on the flags)